### ISSUE TITLE
[1.1.6] Non-static method October\Rain\Database\Model::whereHas() should not be called statically

### ISSUE THREAD

**Issue by fansaien** [octobercms/october#5603]
The function like this:
```php
$posts = Post::whereHas('comments', function ($query) {
    $query->where('content', 'like', 'foo%');
})->get();
``` 
will popup an error: 'Non-static method October\Rain\Database\Model::whereHas() should not be called statically'

The 1.1.5 is working because the `whereHas()` will call `public static function __callStatic($method, $parameters)` in the `Illuminate\Database\Eloquent\Model.php`.

However, the 1.1.6 added `use \October\Rain\Database\Concerns\QueriesRelationships` in the `October\Rain\Database\Model`. Please review https://github.com/octobercms/library/commit/c64184732727b8da4a1a54685e644b154ab55fd5.

The `October\Rain\Database\Concerns` use `\Illuminate\Database\Eloquent\Concerns\QueriesRelationships` and that QueriesRelationships has a non-static function `whereHas`. 

In the 1.1.6, it doesn't call the `__callStatic` method because it already has the wherehas() method.

**Comment by daftspunk** [comment]
This looks like an oversight in the original design, there is no reason that the QueriesRelationships should be implemented in the model (/cc @Klaasie), only in the Builder

So far removing it from the code allows the query to run. We should try this as a first attempt and if there are still issues we can roll it back

Can you try it https://github.com/octobercms/library/commit/fcd3526f0ec19ad1111cfa3db3baa4d553685956?

**Comment by fansaien** [comment]
Yes, removing the `use \October\Rain\Database\Concerns\QueriesRelationships` can fix the non-static method issue, thanks.

**Comment by daftspunk** [comment]
Thanks for reporting back. Fixed in `october/rain 1.1.7`
